check: Issue with cache updated outside from rustic

I recently used `restic migrate upgrade_repo_v2` to modernize an existing Restic repository. The compression that v2 offers is quite attractive. Unfortunately, Rustic does not seem to understand the new format:

```
[ERROR] Error reading file Type: Pack, Id: d78a4373 : No such file or directory (os error 2)
...many more, similar errors follow...
```

Rustic should support v2 repositories as well, so that I can benefit from compression while still reaping the benefits of Rustic's prettier output, better repair tools, etc. Please? :pleading_face:

--- a/src/commands/check.rs
+++ b/src/commands/check.rs
@@ -61,8 +61,12 @@ pub(super) fn execute(repo: OpenRepository, opts: Opts) -> Result<()> {
 
     let index_collector = check_packs(be, hot_be, opts.read_data)?;
 
-    if !opts.trust_cache {
-        if let Some(cache) = &cache {
+    if let Some(cache) = &cache {
+        let p = progress_spinner("cleaning up packs from cache...");
+        cache.remove_not_in_list(FileType::Pack, index_collector.tree_packs())?;
+        p.finish();
+
+        if !opts.trust_cache {
             let p = progress_bytes("checking packs in cache...");
             // TODO: Make concurrency (5) customizable
             check_cache_files(5, cache, raw_be, FileType::Pack, p)?;

--- a/src/commands/prune.rs
+++ b/src/commands/prune.rs
@@ -53,10 +53,6 @@ pub(super) struct Opts {
     #[clap(long)]
     instant_delete: bool,
 
-    /// Only remove unneded pack file from local cache. Do not change the repository at all.
-    #[clap(long)]
-    cache_only: bool,
-
     /// Simply copy blobs when repacking instead of decrypting; possibly compressing; encrypting
     #[clap(long)]
     fast_repack: bool,
@@ -108,20 +104,6 @@ pub(super) fn execute(repo: OpenRepository, opts: Opts, ignore_snaps: Vec<Id>) -
         index_files.push((id, index))
     }
     p.finish();
-
-    if let Some(cache) = &repo.cache {
-        let p = progress_spinner("cleaning up packs from cache...");
-        cache.remove_not_in_list(FileType::Pack, index_collector.tree_packs())?;
-        p.finish();
-    }
-    match (repo.cache.is_some(), opts.cache_only) {
-        (true, true) => return Ok(()),
-        (false, true) => {
-            warn!("Warning: option --cache-only used without a cache.");
-            return Ok(());
-        }
-        _ => {}
-    }
 
     let (used_ids, total_size) = {
         let index = index_collector.into_index();

